bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/ValueDecoder.java

```diff
@@ -0,0 +1,384 @@
+        } catch (KNXException e) {
+            LOGGER.warn("Failed creating a translator for datapoint type '{}'.", dptId, e);
+        }
+
+        return null;
+    }
+
+    private static Type handleDpt1(String subType, DPTXlator translator) {
+        DPTXlatorBoolean translatorBoolean = (DPTXlatorBoolean) translator;
+        switch (subType) {
+            case "008":
+                return translatorBoolean.getValueBoolean() ? UpDownType.DOWN : UpDownType.UP;
+            case "009":
+            case "019":
+                // This is wrong for DPT 1.009. It should be true -> CLOSE, false -> OPEN, but unfortunately
+                // can't be fixed without breaking a lot of working installations.
+                // The documentation has been updated to reflect that. / @J-N-K
+                return translatorBoolean.getValueBoolean() ? OpenClosedType.OPEN : OpenClosedType.CLOSED;
+            case "010":
+                return translatorBoolean.getValueBoolean() ? StopMoveType.MOVE : StopMoveType.STOP;
+            case "022":
+                return DecimalType.valueOf(translatorBoolean.getValueBoolean() ? "1" : "0");
+            default:
+                return OnOffType.from(translatorBoolean.getValueBoolean());
+        }
+    }
+
+    private static @Nullable Type handleDpt3(String subType, DPTXlator translator) {
+        DPTXlator3BitControlled translator3BitControlled = (DPTXlator3BitControlled) translator;
+        if (translator3BitControlled.getStepCode() == 0) {
+            LOGGER.debug("convertRawDataToType: KNX DPT_Control_Dimming: break received.");
+            return UnDefType.NULL;
+        }
+        switch (subType) {
+            case "007":
+                return translator3BitControlled.getControlBit() ? IncreaseDecreaseType.INCREASE
+                        : IncreaseDecreaseType.DECREASE;
+            case "008":
+                return translator3BitControlled.getControlBit() ? UpDownType.DOWN : UpDownType.UP;
+            default:
+                LOGGER.warn("DPT3, subtype '{}' is unknown.", subType);
+                return null;
+        }
+    }
+
+    private static Type handleDpt10(String value) throws ParseException {
+        if (value.contains("no-day")) {
+            /*
+             * KNX "no-day" needs special treatment since openHAB's DateTimeType doesn't support "no-day".
+             * Workaround: remove the "no-day" String, parse the remaining time string, which will result in a
+             * date of "1970-01-01".
+             * Replace "no-day" with the current day name
+             */
+            StringBuilder stb = new StringBuilder(value);
+            int start = stb.indexOf("no-day");
+            int end = start + "no-day".length();
+            stb.replace(start, end, String.format(Locale.US, "%1$ta", Calendar.getInstance()));
+            value = stb.toString();
+        }
+        Date date = null;
```

Comment:
Would you consider skipping the usage of the old Java 8 `Date`? See for example https://github.com/openhab/openhab-addons/pull/14170#discussion_r1063545638

Resulting code from that:
https://github.com/openhab/openhab-addons/blob/596324635d89d293ec9d54100dfde43327ab7273/bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/dto/thermostat/LocalDateTimeDeserializer.java#L42-L47

and:
https://github.com/openhab/openhab-addons/blob/596324635d89d293ec9d54100dfde43327ab7273/bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/dto/thermostat/InstantDeserializer.java#L44-L50

Reply:
Actually I would prefer not to touch this. This is the implementation which is currently in use in the plugin. The test coverage is low, and I am afraid to break something. Maybe this is a cleanup task for later....